File: src/main/java/com/mkreidl/ephemeris/solarsystem/OrbitalModel.java
package com.mkreidl.ephemeris.solarsystem;

import com.mkreidl.ephemeris.Distance;
import com.mkreidl.ephemeris.Time;
import com.mkreidl.ephemeris.geometry.Cartesian;
import com.mkreidl.ephemeris.geometry.Spherical;

public abstract class OrbitalModel
{
    public enum Type
    {
        HELIOCENTRIC, GEOCENTRIC
    }

    public Type getType()
    {
        return Type.HELIOCENTRIC;
    }

    public Distance getDistanceUnit()
    {
        return Distance.AU;
    }

    public abstract void compute( Time time, Cartesian position, Cartesian velocity );

    public abstract void compute( Time time, Spherical position, Spherical velocity );
}

File: src/main/java/com/mkreidl/ephemeris/solarsystem/OrbitalModel.kt
package com.mkreidl.ephemeris.solarsystem

import com.mkreidl.ephemeris.Distance
import com.mkreidl.ephemeris.time.Instant
import com.mkreidl.math.Phase

interface OrbitalModel {

    enum class Type {
        HELIOCENTRIC, GEOCENTRIC
    }

    val type: Type
    val distanceUnit: Distance
    val phase: Phase

    fun compute(instant: Instant): Phase
}
